backend/heartcode/src/main/java/com/itvitae/heartcode/user/UserService.java

```diff
@@ -21,18 +23,27 @@ public Optional<User> findById(String address) {
     return userRepository.findById(address);
   }
 
-  public User save(String email, String alias, String password, UserGender gender) {
+  public User save(
+      String email, String alias, String password, UserGender gender, String dateOfBirthString) {
     if (isInvalidEmail(email) || userWithEmailExists(email)) {
       throw new IllegalArgumentException("email is invalid");
     }
+
     if (alias.isBlank()) {
       throw new IllegalArgumentException("alias is invalid");
     }
+
     if (password.isBlank()) {
       throw new IllegalArgumentException("password is invalid");
     }
 
-    return userRepository.save(new User(email, alias, passwordEncoder.encode(password), gender));
```

Comment:
ik zie dat dit niet wordt gecheckt in de controller, dit levert dus een 500 Internal Server Error op als iemand een onjuiste date invult, de parseDateOfBirth moet dus eigenlijk in de controller gebeuren, dan kun je dat daar valideren en een 401 terugsturen als de dateOfBirth niet klopt